### PR TITLE
Add TestData project containing Unicode UCD files

### PR DESCRIPTION
**Pull request by GrabYourPitchforks**
This contains the Unicode data files that are needed by https://github.com/dotnet/runtime/pull/328. I followed steps 1 - 3 in [the README](https://github.com/dotnet/runtime-assets/blob/master/README.md) and verified that this package can be consumed by our test projects.

Open questions:

 1. Do we really want to name this `Private.CoreFx.TestUtilities.Unicode.TestData`? It will eventually be consumed by several unit projects, including _Private.CoreFx.TestUtilities.Unicode_, _System.Globalization.Tests_, and _System.Text.Encodings.Web.Tests_. (I'll need to make a minor change to this and bump the version to 1.0.1 before that last project can consume it.)

 2. Do we need `THIRD-PARTY-NOTICES.TXT`? These files are copied from https://www.unicode.org/Public/, and we have a similar notices file in the _runtime_ repo.

### CODE REVIEW

**Comment by ViktorHofer** [comment]
Yes we need a THIRD-PARTY-NOTICES.TXT. Please see https://github.com/dotnet/runtime-assets/commit/2a901883900fc109dba40c916ad2109906fc7ebc for an example.

> Do we really want to name this Private.CoreFx.TestUtilities.Unicode.TestData?

You decide on the name :)

**Comment by GrabYourPitchforks** [comment]
@ViktorHofer I didn't even see that notices file under the project folder. Should I also throw this one under the new project folder (which I guess means it gets swept up as part of the nupkg?), or do we want to move it to the repo root?

**Comment by ViktorHofer** [comment]
I think we should move it to the repo root and include your notice. I think it being part of the package was a mistake.

**Comment by GrabYourPitchforks** [comment]
Ok, I'll fix it up when I'm back in the office.

**Comment by GrabYourPitchforks** [comment]
Changes in most recent iteration:

 * Old `THIRD-PARTY-NOTICES.TXT` merged into new file at repo root.
 * Bumped version of `System.IO.Compression.TestData` since I technically changed the contents of the package.
 * Renamed `CoreFx.Private.TestUtilities.Unicode.TestData` to `Microsoft.Private.UnicodeData`.

**Comment by ViktorHofer** [comment]
What about `Microsoft.Private.Runtime.UnicodeData` to make the intent clear?

File: runtime-assets.sln [hunk 1]
```diff
@@ -19,6 +19,8 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "System.Text.RegularExpressi
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "System.Windows.Extensions.TestData", "src\System.Windows.Extensions.TestData\System.Windows.Extensions.TestData.csproj", "{BB874A4E-3388-44EB-B4DC-40C0F38572C8}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "CoreFx.Private.TestUtilities.Unicode.TestData", "CoreFx.Private.TestUtilities.Unicode.TestData\CoreFx.Private.TestUtilities.Unicode.TestData.csproj", "{6E7ACB67-F024-4289-A86E-482A88BEE7D5}"
```
Comment (ViktorHofer) [inline runtime-assets.sln]: That path and name needs to be updated.

File: src/Microsoft.Private.UnicodeData/Microsoft.Private.UnicodeData.csproj [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <PackageVersion>1.0.0</PackageVersion>
```
Comment (ViktorHofer) [inline src/Microsoft.Private.UnicodeData/Microsoft.Private.UnicodeData.csproj]: you can remove that version now, as we switch to the dependency flow system with automatic versioning.

File: src/System.IO.Compression.TestData/System.IO.Compression.TestData.csproj [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <PackageVersion>1.0.15</PackageVersion>
+    <PackageVersion>1.0.16</PackageVersion>
```
Comment (ViktorHofer) [inline src/System.IO.Compression.TestData/System.IO.Compression.TestData.csproj]: same here, pls remove.

File: src/Microsoft.Private.Runtime.UnicodeData/Microsoft.Private.Runtime.UnicodeData.csproj [hunk 4]
```diff
@@ -0,0 +1,2 @@
+<Project Sdk="Microsoft.NET.Sdk">
+</Project>
```
Comment (ViktorHofer) [inline src/Microsoft.Private.Runtime.UnicodeData/Microsoft.Private.Runtime.UnicodeData.csproj]: As this is not a normal TestData project you need to specify these properties manually: https://github.com/dotnet/runtime-assets/blob/ff5202ce3712c3111586d84c1670c947df65ec63/Directory.Build.props#L32-L34.

File: runtime-assets.sln [hunk 5]
```diff
@@ -3,6 +3,8 @@ Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
 VisualStudioVersion = 15.0.26124.0
 MinimumVisualStudioVersion = 15.0.26124.0
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.Private.Runtime.UnicodeData", "src\Microsoft.Private.Runtime.UnicodeData\Microsoft.Private.Runtime.UnicodeData.csproj", "{6E7ACB67-F024-4289-A86E-482A88BEE7D5}"
```
Comment (akoeplinger) [inline runtime-assets.sln]: I'd prefer `System.Private.Runtime.UnicodeData` for the name, unless there's some precedent for having the `Microsoft.Private` convention.

**Comment by GrabYourPitchforks** [comment]
_System.Private.*_ works for me. :)

**Pull request merged**